satpy/writers/cf/time.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2017-2023 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""CF processing of time dimension and coordinates."""
+import logging
+
+import numpy as np
+import xarray as xr
+
+logger = logging.getLogger(__name__)
+
+
+EPOCH = u"seconds since 1970-01-01 00:00:00"
+
+
+def add_time_bounds_dimension(ds, time="time"):
+    """Add time bound dimension to xr.Dataset."""
+    start_times = []
+    end_times = []
+    for _var_name, data_array in ds.items():
+        start_times.append(data_array.attrs.get("start_time", None))
+        end_times.append(data_array.attrs.get("end_time", None))
+
+    start_time = min(start_time for start_time in start_times
+                     if start_time is not None)
+    end_time = min(end_time for end_time in end_times
+                   if end_time is not None)
+    ds['time_bnds'] = xr.DataArray([[np.datetime64(start_time),
+                                     np.datetime64(end_time)]],
+                                   dims=['time', 'bnds_1d'])
+    ds[time].attrs['bounds'] = "time_bnds"
+    ds[time].attrs['standard_name'] = "time"
+    return ds
+
+
+def _process_time_coord(dataarray, epoch):
```

Comment:
While this is imported by `datarray.make_cf_dataarray`. 
Maybe this function could go in a `attrs_coords.py` file together with the content of the `coords_attrs.py` which currently add attributes to the x,y coordinates. 
In order to separate stuffs which are used for DataArray and Dataset generation.

Reply:
What are the functions all doing that would go in `attrs_coords.py`? They are creating or manipulating `coords` variables, right? Can it just be called `coords.py`?